src/main/java/org/apache/commons/io/build/AbstractOriginSupplier.java

```diff
@@ -368,4 +376,26 @@ public B setURI(final URI origin) {
     public B setWriter(final Writer origin) {
         return setOrigin(newWriterOrigin(origin));
     }
+
+    /**
+     * Sets a new origin.
+     *
+     * @param origin the new origin.
+     * @return {@code this} instance.
+     * @since 2.21.0
+     */
+    public B setReadableByteChannel(final ReadableByteChannel origin) {
+        return setOrigin(newChannelOrigin(origin));
+    }
+
+    /**
+     * Sets a new origin.
+     *
+     * @param origin the new origin.
+     * @return {@code this} instance.
+     * @since 2.21.0
+     */
+    public B setWritableByteChannel(final WritableByteChannel origin) {
```

Comment:
If `newChannelOrigin(Channel)` and `ChannelOrigin.ChannelOrigin(Channel)` both take a Channel, why not simply have `setChannel(Channel)`?

Reply:
In https://github.com/apache/commons-io/pull/784/commits/c4e3b4300f2221ccfbcb14adf5184c7be5154b8d I replaced the `getReadableByteChannel` and similar methods with those proposed above: the API looks much cleaner now.